diagnostic/handlers.go

```diff
@@ -0,0 +1,582 @@
+package diagnostic
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/influxdata/influxdb/coordinator"
+	"github.com/influxdata/influxdb/monitor"
+	"github.com/influxdata/influxdb/query"
+	"github.com/influxdata/influxdb/services/collectd"
+	"github.com/influxdata/influxdb/services/continuous_querier"
+	"github.com/influxdata/influxdb/services/graphite"
+	"github.com/influxdata/influxdb/services/httpd"
+	"github.com/influxdata/influxdb/services/meta"
+	"github.com/influxdata/influxdb/services/opentsdb"
+	"github.com/influxdata/influxdb/services/precreator"
+	"github.com/influxdata/influxdb/services/retention"
+	"github.com/influxdata/influxdb/services/snapshotter"
+	"github.com/influxdata/influxdb/services/subscriber"
+	"github.com/influxdata/influxdb/services/udp"
+	"github.com/uber-go/zap"
+)
+
+type CollectdHandler struct {
+	l zap.Logger
+}
+
+func (s *Service) InCollectdContext() collectd.Diagnostic {
+	if s == nil {
+		return nil
+	}
+	return &CollectdHandler{l: s.l.With(zap.String("service", "collectd"))}
+}
+
+func (h *CollectdHandler) Starting() {
+	h.l.Info("Starting collectd service")
+}
+
+func (h *CollectdHandler) Listening(addr net.Addr) {
+	h.l.Info(fmt.Sprint("Listening on UDP: ", addr.String()))
+}
+
+func (h *CollectdHandler) Closed() {
+	h.l.Info("collectd UDP closed")
+}
+
+func (h *CollectdHandler) UnableToReadDirectory(path string, err error) {
+	h.l.Info(fmt.Sprintf("Unable to read directory %s: %s\n", path, err))
+}
+
+func (h *CollectdHandler) LoadingPath(path string) {
+	h.l.Info(fmt.Sprintf("Loading %s\n", path))
+}
+
+func (h *CollectdHandler) TypesParseError(path string, err error) {
+	h.l.Info(fmt.Sprintf("Unable to parse collectd types file: %s\n", path))
```

Comment:
Any reason for not using `zap.Fields` in each of these methods?

Something along the line of 

```go
h.l.Info("Unable to parse collectd types file", zap.String("path", path))
```

Reply:
No reason at the moment. Right now, I'm just doing pretty straight translation. We only ever half did structured logging in InfluxDB. I moved the code to a structured logger, but used `Sprintf` so basically nothing else changed. The intention was to then have follow up commits that moved to structured logging, but then we couldn't figure out how we wanted to proceed on the logging front.

Reply:
I see. That makes sense.